src/zope/configuration/tests/test_config.py

```diff
@@ -114,6 +114,23 @@ def test_resolve_bad_sub_import(self):
            if name in sys.modules:
                del sys.modules[name]
 
+    def test_resolve_unicode_all(self):
+        # If a package's __init__.py is in the process of being ported from
+        # Python 2 to Python 3 using unicode_literals, then it can end up
+        # with unicode items in __all__, which breaks star imports from that
+        # package; but we tolerate this.
+        import sys
+        c = self._makeOne()
+        self.assertEqual(
+            c.resolve('zope.configuration.tests.unicode_all').foo, 'sentinel')
+        self.assertEqual(
+            c.resolve('zope.configuration.tests.unicode_all.foo'), 'sentinel')
+        # Cleanup:
+        for name in ('zope.configuration.tests.unicode_all',
+                     'zope.configuration.tests.unicode_all.__future__'):
+            if name in sys.modules:
+                del sys.modules[name]
```

Comment:
I would prefer the cleanup to be in a `tearDown` method (or implemented via `self.addCleanup(sys.modules.pop, 'zope....', None)` maybe).

TBH I'm not sure cleanup is necessary here.  We don't usually consider actual modules getting imported into sys.modules to be a side effect worth cleaning up.

Reply:
This was clone-and-hack from other tests.  I guess that it's relevant in those cases because multiple test cases try to import the same test modules, and that won't work so well without the cleanup?

Reply:
Consistency with other tests is a good-enough reason.